desktop/src/ui.rs

```diff
@@ -1,12 +1,75 @@
+use chrono::{DateTime, Utc};
 use clipboard::{ClipboardContext, ClipboardProvider};
-use ruffle_core::backend::ui::{MouseCursor, UiBackend};
+use rfd::{AsyncFileDialog, FileHandle};
+use ruffle_core::backend::ui::{
+    DialogResultFuture, Error, FileDialogResult, FileFilter, MouseCursor, UiBackend,
+};
 use ruffle_core::events::{KeyCode, PlayerEvent};
 use std::collections::HashSet;
+use std::fs::{metadata, Metadata};
 use std::rc::Rc;
 use tinyfiledialogs::{message_box_ok, MessageBoxIcon};
 use winit::event::{ElementState, ModifiersState, VirtualKeyCode, WindowEvent};
 use winit::window::Window;
 
+pub struct DesktopFileDialogResult {
+    handle: Option<FileHandle>,
+    md: Option<Metadata>,
+}
+
+impl DesktopFileDialogResult {
+    pub fn new(handle: Option<FileHandle>) -> Self {
+        let md = handle.as_ref().and_then(|x| metadata(x.path()).ok());
```

Comment:
style nit: Let's import `fs::self` and then do `fs::metadata` (makes it a little clearer that it's the standard metadata fn and not our own fn.)